### PR TITLE
Add finetune scripts

### PR DESCRIPTION
**Pull request by MariusBBerg**
## Summary
This PR adds the fine-tuning script work on `finetune-scripts` and cleans it up for codebase readiness.

It includes:
- adding new finetune utility scripts for evaluation and dataset reversal
- updating the existing dataset split and training scripts
- refactoring the finetune scripts for readability and maintainability
- removing low-value comments and normalizing structure, helper boundaries, and formatting
- preserving the existing intended functionality and script usage

### CODE REVIEW

**Review by sebalexsmith — APPROVED** [review]
Looks good.

**Review by Marstar03 — APPROVED** [review]
